src/modules/rlm_json/json.c

```diff
@@ -252,3 +252,105 @@ void fr_json_version_print(void)
 #endif
 }
 
+static json_object *add_vp_to_json_array(TALLOC_CTX *ctx, json_object *arr, VALUE_PAIR *vp) {
+	json_object *to_add = NULL;
+	char *tmp;
+	size_t len, outlen;
+
+	switch (vp->da->type) {
+		case PW_TYPE_INTEGER:
+			to_add = json_object_new_int(vp->vp_integer);
+			break;
+		case PW_TYPE_SHORT:
+			to_add = json_object_new_int(vp->vp_short);
+			break;
+		case PW_TYPE_BYTE:
+			to_add = json_object_new_int(vp->vp_byte);
+			break;
+		case PW_TYPE_SIGNED:
+			to_add = json_object_new_int(vp->vp_signed);
+			break;
+		case PW_TYPE_INTEGER64:
+			to_add = json_object_new_int64(vp->vp_integer64);
+			break;
+		case PW_TYPE_BOOLEAN:
+			to_add = json_object_new_boolean(vp->vp_byte);
+			break;
+		default:
+			len = vp->data.length+1;
+			tmp = NULL;
+			do {
+				if (tmp) talloc_free(tmp);
+				tmp = talloc_array(ctx, char, len);
+				if (!tmp) return NULL;
+				outlen = fr_pair_value_snprint(tmp, len, vp, 0);
+				len <<= 1;
+			} while (outlen >= len);
+			to_add = json_object_new_string(tmp);
+			talloc_free(tmp);
+			break;
+	}
+
+	if (to_add)
+		json_object_array_add(arr, to_add);
+
+	return to_add;
+}
+
+const char *fr_json_vps_to_json(TALLOC_CTX *ctx, VALUE_PAIR **vps, const char *prefix) {
```

Comment:
That's why we use from and to, so you can have function names that make sense without duplicating bits